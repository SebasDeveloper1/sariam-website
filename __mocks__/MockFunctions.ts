import '@testing-library/jest-dom/extend-expect';
export const MockFunction = jest.fn();
