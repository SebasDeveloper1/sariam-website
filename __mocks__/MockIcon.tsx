// Mock the startIcon and endIcon components for testing
export const MockIcon = (): JSX.Element => <div id="mock-icon">Icon</div>;
