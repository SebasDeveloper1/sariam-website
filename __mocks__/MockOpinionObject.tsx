import { MockOpinionTestProps } from '@/components/OpinionCard/OpinionCard.model'; // Mocked opinion data for testing
export const MockOpinion: MockOpinionTestProps = {
  author: {
    name: 'John Doe',
    role: 'CEO',
    photo: {
      url: 'https://example.com/photo.jpg',
      description: 'John Doe photo',
    },
  },
  content: 'This is a great product!',
  sys: { id: 'HUWhn2yQFFqvsAKxX6Htl' },
};
