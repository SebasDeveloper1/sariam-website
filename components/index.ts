import Button from './Buttons/Button/Button';
import IconButton from './Buttons/IconButton/IconButton';
import SliderComponent from './SliderComponent/SliderComponent';
import ServiceCard from './ServiceCard/ServiceCard';
export { Button, IconButton, SliderComponent, ServiceCard };
