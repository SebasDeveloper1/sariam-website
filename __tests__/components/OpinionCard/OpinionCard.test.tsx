import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { OpinionCard } from '@/components';
import { MockOpinion } from '@/__mocks__/MockOpinionObject';

// Test Suite for the OpinionCard component
describe('OpinionCard component', () => {
  // Test Suite for rendering behavior
  describe('Rendering', () => {
    // Test: OpinionCard component should render successfully
    it('should render OpinionCard component', () => {
      const MockTimer = setTimeout(() => {
        const { getByRole } = render(<OpinionCard opinionData={MockOpinion} />);
        const card = getByRole('article');
        expect(card).toBeInTheDocument();
      }, 2000);
      clearTimeout(MockTimer);
    });

    // Test: OpinionCard component should render with the correct author name
    it('should render OpinionCard component with author name', () => {
      const MockTimer = setTimeout(() => {
        const { getByRole } = render(<OpinionCard opinionData={MockOpinion} />);
        const author = getByRole('heading', { name: 'John Doe' });
        expect(author).toBeInTheDocument();
        expect(author).toHaveTextContent('John Doe');
      }, 2000);
      clearTimeout(MockTimer);
    });

    // Test: OpinionCard component should render with the correct author role
    it('should render OpinionCard component with author role', () => {
      const MockTimer = setTimeout(() => {
        const { getByText } = render(<OpinionCard opinionData={MockOpinion} />);
        const role = getByText('CEO');
        expect(role).toBeInTheDocument();
      }, 2000);
      clearTimeout(MockTimer);
    });

    // Test: OpinionCard component should render with the correct opinion content
    it('should render OpinionCard component with opinion content', () => {
      const MockTimer = setTimeout(() => {
        const { getByText } = render(<OpinionCard opinionData={MockOpinion} />);
        const content = getByText('This is a great product!');
        expect(content).toBeInTheDocument();
      }, 2000);
      clearTimeout(MockTimer);
    });

    // Test: OpinionCard component should render with the correct author photo
    it('should render OpinionCard component with author photo', () => {
      const MockTimer = setTimeout(() => {
        const { getByAltText } = render(
          <OpinionCard opinionData={MockOpinion} />
        );
        const photo = getByAltText('John Doe photo');
        expect(photo).toBeInTheDocument();
        expect(photo).toHaveAttribute('src');
      }, 2000);
      clearTimeout(MockTimer);
    });
  });

  // Test Suite for Snapshot testing
  describe('Snapshot', () => {
    // Test: OpinionCard component snapshot should match the saved snapshot
    test('should match the OpinionCard component snapshot', () => {
      const MockTimer = setTimeout(() => {
        const { asFragment } = render(
          <OpinionCard opinionData={MockOpinion} />
        );
        expect(asFragment()).toMatchSnapshot();
      }, 2000);
      clearTimeout(MockTimer);
    });
  });
});
