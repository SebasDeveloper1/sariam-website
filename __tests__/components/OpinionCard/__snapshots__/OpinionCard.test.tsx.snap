// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpinionCard component Snapshot should match the OpinionCard component snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden flex flex-col justify-start items-center gap-y-4 w-full p-6 rounded-2xl bg-white shadow-xl shadow-slate-900/10"
    role="article"
  >
    <figcaption
      class="flex justify-between items-center gap-x-3 w-full"
    >
      <figure
        class="relative overflow-hidden w-14 aspect-square rounded-full object-cover"
      >
        <img
          alt="John Doe photo"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="https://example.com/photo.jpg?w=256&h=256&fit=fill"
          srcset="https://example.com/photo.jpg?w=128&h=128&fit=fill 1x, https://example.com/photo.jpg?w=256&h=256&fit=fill 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="100"
        />
      </figure>
      <div
        class="w-full"
      >
        <h6
          class="heading_6 font-semibold text-gray-900 capitalize leading-none"
        >
          John Doe
        </h6>
        <span
          class="span_sm font-medium text-lime-600 capitalize"
        >
          CEO
        </span>
      </div>
    </figcaption>
    <div
      class="w-full"
    >
      <blockquote
        class="overflow-hidden w-full paragraph_base text-gray-700"
      >
        This is a great product!
      </blockquote>
    </div>
  </div>
</DocumentFragment>
`;
